### PR TITLE
feat: use subgraph to pre-populate gauge options for the safe user in ui

### PR DESCRIPTION
**Pull request by petrovska-petro**

### CODE REVIEW

File: src/components/GraphQuery.tsx
Comment (gosuto-inzasheru) [inline src/components/GraphQuery.tsx:1]: i would opt for deleting this file, this was just a poc for me. no real value in this really

File: src/queries/useGetUserGaugePositions.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        ) {
+          gauge {
+            id
+            protocol
+            pool
+            pool_name
+          }
+        }
+      }
+    `;
```
Comment (gosuto-inzasheru) [inline src/queries/useGetUserGaugePositions.tsx:33]: so didnt you want to retrieve this from the separate `user_active_stakes.graphql` file? i liked this idea, even if it is just for syntax and maintenance.

File: src/queries/useGetUserGaugePositions.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    return gaugeResults;
+  } catch (error) {
+    console.error(error);
+    return null;
+  }
+}
+
+export default function useGetUserGaugePositions() {
+  const { address } = useAccount();
+  return useQuery(["userGaugePositions", address?.toLowerCase()], () =>
```
Comment (gosuto-inzasheru) [inline src/queries/useGetUserGaugePositions.tsx:52]: yes to lower case is important here, all addresses in subgraphs are saved lowercase!

File: package.json [hunk 3]
```diff
@@ -26,6 +26,8 @@
     "@tanstack/react-table": "^8.9.3",
     "@wagmi/cli": "^1.3.0",
     "buffer": "^6.0.3",
+    "graphql": "^16.8.0",
```
Comment (gosuto-inzasheru) [inline package.json:29]: is this a req for `graphql-request`?

**Review by gosuto-inzasheru — COMMENTED** [review]
looks great, works on localhost for me!

Screenshot 2023-08-23 at 09 39 27

seeing the pool name of the gauge and not the address is a great step in ux :)

this does raise a new question; what if we want to enable harvests for both?

and probably also somewhat out of scope, but i noted that even if connected to opti goerli the same gauges show up. so somewhere we should start becoming chain aware. (a subgraph will always be for a single specific chain only, for a different chain a different subgraph will be needed)

File: package.json (hunk 3 above)
Comment (gosuto-inzasheru) [reply package.json:29]: ah yea looks like it: https://www.npmjs.com/package/graphql-request#why-do-i-have-to-install-graphql

File: src/components/GraphQuery.tsx
Comment (petrovska-petro) [reply src/components/GraphQuery.tsx:1]: alright, removed in 030cdf7fd2b7543d37891eb9aed4a302d3b69a2f

File: src/queries/useGetUserGaugePositions.tsx (hunk 1 above)
Comment (petrovska-petro) [reply src/queries/useGetUserGaugePositions.tsx:33]: true, i will like to keep it on that way, will update it and hopefully can add also the chain-awareness

Comment (gosuto-inzasheru) [reply src/queries/useGetUserGaugePositions.tsx:33]: #33

**Comment by petrovska-petro** [comment]
> and probably also somewhat out of scope, but i noted that even if connected to opti goerli the same gauges show up. so somewhere we should start becoming chain aware. (a subgraph will always be for a single specific chain only, for a different chain a different subgraph will be needed)

added the chain awareness in 9249a3041ccf1c21823b747f62fb93a0656228c6

**Comment by petrovska-petro** [comment]
> this does raise a new question; what if we want to enable harvests for both?

another item was planning to scope on the ui side is to allow multi-select on the options and to do a multi-call in the safe app to enable both atomically

File: src/queries/useGetUserGaugePositions.tsx (hunk 1 above)
Comment (petrovska-petro) [reply src/queries/useGetUserGaugePositions.tsx]: i extracted the gql query into its separate directory for the time being, see be555d955c8a0578953bd8b009426172ed1043de

then, i think we can spend some more time in #33 to explore the right structure/architecture and if its possible to auto-generate all this helpers out of the queries itself

**Review by gosuto-inzasheru — COMMENTED** [review]
i need to refresh after switching chains, in order to refresh the gauge dropdown

maybe open a separate issue for this?

**Comment by gosuto-inzasheru** [comment]
because the rest lgtm and can be merged imo :)

**Review by Gb2real — APPROVED** [review]
Be great

**Comment by Gb2real** [comment]
Im happy

**Pull request merged**